working/macros/feature-specification.md

```diff
@@ -606,17 +582,24 @@ of a function or initializer for a variable. It is encouraged to provide a body
 (or initializer) if possible, but you can opt to wait until the definition phase
 if needed.
 
-When applied to a class, enum, or mixin a macro in this phase can introspect on
-all of the members of that class and its superclasses, but it cannot introspect
-on the members of other types. For mixins, the `on` type is considered a
-superclass and is introspectable. Note that generic type arguments are not
-introspectable.
-
-When applied to an extension, a macro in this phase can introspect on all of the
-members of the `on` type, as well as its generic type arguments and the bounds
-of any generic type parameters for the extension.
-
-TODO: Define the introspection rules for extension types.
+Phase two macros can introspect on all of the members of a type. If the type
+to be introspected is declared in the same library cycle and has one or more
+macros applied to it then this introduces an ordering constraint between the
+macro applications: the introspection call waits for complete results before
+returning, meaning it waits for the macro applications on the target type to
+finish.
+
+If a cycle arises in macro applications waiting for other macro applications to
+complete then a special value is returned indicating that introspection is not
```

Comment:
I do think it makes sense to throw an error here, we can't easily model a special value otherwise (I don't want to go down the path of Option types or something).

Reply:
Fwiw I am totally fine with making this an Error instead of an Exception, to indicate that it should not be caught/handled? I don't think many macros would be able to actually do anything meaningful.

I mostly just think the easiest way to surface this error (and terminate the macro), is to just have the API throw. I don't think we really have to worry if some macro author does actually try to handle the error, they can't do anything _harmful_, its just that they probably can't do anything _useful_ (but maybe some could, who knows).

Reply:
Sounds good--how about like this? (just throw `StateError`).

Thanks.